lib/segment/src/index/hnsw_index/graph_layers.rs

```diff
@@ -241,20 +240,24 @@ impl GraphLayers {
     }
 
     fn num_points(&self) -> usize {
-        self.links_layers.len()
+        self.links.num_points()
     }
 
     pub fn point_level(&self, point_id: PointOffsetType) -> usize {
-        self.links_layers[point_id as usize].len() - 1
+        self.links.point_level(point_id)
     }
 
     pub fn merge_from_other(&mut self, other: GraphLayers) {
+        self.max_level = std::cmp::max(self.max_level, other.max_level);
```

Comment:
De-compressing flatten representation might be expensive in the scenario, when we need to merge a lot of small graphs (the common scenario, actually). Let's only use merge on regular graph representation and compress only the final version of it

Reply:
fixed